src/source/Sctp/Sctp.h

```diff
@@ -59,11 +53,21 @@ typedef struct {
 } SctpSessionCallbacks, *PSctpSessionCallbacks;
 
 typedef struct {
-    volatile SIZE_T shutdownStatus;
     struct socket *socket;
     SctpSessionCallbacks sctpSessionCallbacks;
+    UINT64 key;
```

Comment:
So you use store the current time here and this acts as the key in the hash table. Perhaps add a comment to state what you use as a key. Also, we really need to create a hashset object in PIC